container/src/lib.rs

```diff
@@ -86,20 +87,23 @@ impl Container {
     pub fn run(&self) -> Result<()> {
         let mounts = self.mounts.clone();
         let code = unsafe {
-            unshare::Command::from(&self.command)
+            let mut child = match unshare::Command::from(&self.command)
                 .chroot_dir(&self.rootfs)
                 .unshare(&*self.namespaces.get())
                 .pre_exec(move || Mounts::apply(&mounts))
                 .envs(self.environment.get())
                 .spawn()
-                .map_err(Error::ContainerSpawnCommand)?
-                .wait()
-                .map_err(Error::ContainerWaitCommand)?
-                .code()
+            {
+                Ok(child) => child,
+                Err(_) => {
+                    let _ = &self.mounts.cleanup(self.rootfs.clone())?;
+                    exit(2);
+                }
+            };
+            child.wait().map_err(Error::ContainerWaitCommand)?.code()
         };
 
         let _ = &self.mounts.cleanup(self.rootfs.clone())?;
```

Comment:
I believe we should be able to just do `self.mounts.cleanup(self.rootfs.clone())?;`here.